src/azul/chalice.py

```diff
@@ -504,34 +497,122 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    # noinspection PyPep8Naming
+    @attr.define(frozen=True)
+    class metric_alarm:
+        """
+        Use this decorator to configure a Lambda's metric alarm, specify a
+        custom number of Lambda invocation resulting in a failure or throttling
+        (threshold) and for which period (5 min - 1 day) are necessary to
+        trigger the metric alarm for the decorated event handler function. This
+        decorator cannot be used to decorate view functions, i.e. functions also
+        decorated with ``@app.route``.
+
+        """
+        #: The AWS/Lambdas namespace for this alarm
+        metric: LambdaMetric
+        #: The value used by the alarm, that when exceeded triggers the alarm
+        threshold: int = attr.ib(default=0)
+        #: The time in seconds to evaluate the alarm threshold
+        period: int = attr.ib(default=5 * 60)
+        lambda_name: Optional[str] = attr.ib(default=None)
+        handler_name: Optional[str] = attr.ib(default=None)
+
+        def __call__(self, f):
+            assert isinstance(f, chalice.app.EventSourceHandler), f
+            try:
+                metric_alarms = getattr(f, 'metric_alarms')
+            except AttributeError:
+                metric_alarms = f.metric_alarms = []
+            metric_alarms.append(self)
+            return f
+
+        def bind(self, lambda_name: str, handler_name: Optional[str] = None, /) -> Self:
+            return attr.evolve(self,
+                               lambda_name=lambda_name,
+                               handler_name=handler_name)
+
+        @property
+        def lambda_tf_resource_name(self) -> str:
+            if self.handler_name is None:
+                return self.lambda_name
+            else:
+                assert self.handler_name != ''
+                return f'{self.lambda_name}_{self.handler_name}'
+
+        @property
+        def alarm_tf_resource_name(self) -> str:
+            return f'{self.lambda_tf_resource_name}_{self.metric.name}'
+
+    # noinspection PyPep8Naming
+    @attr.define(frozen=True)
+    class retry:
+        """
+        Use this decorator to specify the number of times a Lambda invocation of
+        the decorated event handler function should be retried. This decorator
+        cannot be used to decorate view functions, i.e. functions also decorated
+        with ``@app.route``.
+
+        https://docs.aws.amazon.com/lambda/latest/dg/invocation-retries.html
+        """
+        value: int
```

Comment:
More descriptive name might be nice
```suggestion
        times: int
```